pyvista/core/pyvista_ndarray.py

```diff
@@ -0,0 +1,111 @@
+"""Contains pyvista_ndarray a numpy ndarray type used in pyvista."""
+
+import numpy
+from pyvista.utilities.helpers import convert_array, FieldAssociation
+from vtk.numpy_interface.dataset_adapter import VTKObjectWrapper, VTKArray
+from vtk.vtkCommonCore import vtkWeakReference
+from vtk.vtkCommonKitPython import vtkDataArray, vtkAbstractArray
+
+
+class pyvista_ndarray(VTKArray):
```

Comment:
What about pyvista_array? @akaszynski thoughts on this? I generally don't like shortening things in this way so I'm split on this.

Reply:
I vote `pvarray`. Similar format to `ndarray` and get the point across.

Reply:
I originally liked the `*_ndarray` naming just because it implied that it extends the NumPy `ndarray` class. Now we are wrapping `VTKArray` but that is technically still a subclass of `numpy.ndarray` so I'd like to keep that aspect of the name for clarity (make sure people realize that those are indeed numpy arrays as they are passed back to user quite often).

I also prefer `pyvista_*` over `pv_*` - it follows our convention of not aliasing pyvista within the API and only using the `pv` alias on the user side

Reply:
any name for this is really fine by me though! It's rarely directly interacted with by the user

Reply:
Eh, I'll leave it for now. It's only used right now(in master at least) in 1 place, if someone decides it should be changed later on that's fine.